v2/internal/controllers/crd_networking_applicationgateway_test.go

```diff
@@ -0,0 +1,389 @@
+			AddressPrefix: to.Ptr("10.0.0.0/24"),
+			AzureName:     "AzureAppGtwSubnet", // The subnet must have this name in Azure for Bastion to work
+		},
+	}
+	publicIPAddress := newPublicIp(tc, testcommon.AsOwner(rg))
+	tc.CreateResourceAndWait(publicIPAddress)
+	tc.CreateResourceAndWait(vnet)
+	tc.Expect(vnet.Status.Id).ToNot(BeNil())
+	tc.CreateResourceAndWait(subnet)
+	tc.Expect(subnet.Status.Id).ToNot(BeNil())
+	appGtsListnerName := "app-gw-http-listner-1"
+	httpListnerParams := map[string]string{
+		"param-key1": "applicationGateways",
+		"param-val1": appGatewayName,
+		"param-key2": "httpListeners",
+		"param-val2": appGtsListnerName,
+	}
+	appGtwFHttpListnerID, err := getFrontendPortsARMID(tc, rg, httpListnerParams)
+	tc.Expect(err).To(BeNil())
+	appGtwFIPConfig := defineApplicationGatewayIPConfiguration(tc.MakeReferenceFromResource(subnet))
+	appGtwFeIpConfig, appGtwFeIpConfigID := defineApplicationGatewayFrontendIPConfiguration(tc, rg, appGatewayName, tc.MakeReferenceFromResource(subnet), tc.MakeReferenceFromResource(publicIPAddress))
+	appGtwFEPorts, appGtwFePortsID := defineApplicationGatewayFrontendPort(tc, rg, appGatewayName)
+	appGtwSslCerts, appGtwSslCertsID := defineApplicationGatewaySslCertificate(tc, rg, appGatewayName)
+	appGtwSslProfiles, appGtwSslProfilesID := defineApplicationGatewaySslProfile(tc, rg, appGatewayName)
+	appGtwBackendPools, appGtwBackendPoolsID := defineApplicationGatewayBackendAddressPool(tc, rg, appGatewayName)
+	appGtwBackendHttpSettings, appGtwBackendHttpSettingsID := defineApplicationGatewayBackendHttpSettings(tc, rg, appGatewayName)
+	appGtwWafConfig, appGtwWafConfigID := defineApplicationGatewayWebApplicationFirewallConfiguration(tc, rg, appGatewayName)
+	log.Println(appGtwSslProfilesID)
+	log.Println(appGtwWafConfigID)
+	applicationGateway := &network.ApplicationGateway{
+		ObjectMeta: tc.MakeObjectMetaWithName(appGatewayName),
+		Spec: network.ApplicationGateway_Spec{
+			AutoscaleConfiguration: &network.ApplicationGatewayAutoscaleConfiguration{
+				MaxCapacity: to.Ptr[int](3),
+				MinCapacity: to.Ptr[int](1),
+			},
+			AzureName: appGatewayName,
+			Location:  to.Ptr[string](rg.Location()),
+			Owner:     testcommon.AsOwner(rg),
+			Sku: &network.ApplicationGatewaySku{
+				Name: to.Ptr[network.ApplicationGatewaySku_Name](network.ApplicationGatewaySku_Name(network.ApplicationGatewaySku_Name_STATUS_WAF_V2)),
+				Tier: to.Ptr[network.ApplicationGatewaySku_Tier](network.ApplicationGatewaySku_Tier(network.ApplicationGatewaySku_Tier_STATUS_WAF_V2)),
+			},
+			GatewayIPConfigurations:  appGtwFIPConfig,
+			FrontendIPConfigurations: appGtwFeIpConfig,
+
+			FrontendPorts:   appGtwFEPorts,
+			SslCertificates: appGtwSslCerts,
+			SslProfiles:     appGtwSslProfiles,
+			SslPolicy: to.Ptr[network.ApplicationGatewaySslPolicy](network.ApplicationGatewaySslPolicy{
+				PolicyName: to.Ptr[network.PolicyNameEnum](network.PolicyNameEnum(network.PolicyNameEnum_AppGwSslPolicy20220101S)),
```

Comment:
same, don't need `[network.PolicyNameEnum]`, it should be inferred you can just delete the bit in the square brackets